octoprint_mrbeam/mrb_logger.py

```diff
@@ -158,12 +164,23 @@ def _analytics_log_event(self, level, msg, analytics_id, *args, **kwargs):
         if analytics_handler is not None:
             try:
                 msg = msg % args if args and msg else msg
-                caller = None
-                caller_myself = getframeinfo(stack()[0][0])
+                caller = getframeinfo(stack()[0][0])
+                caller_myself = caller
                 i = 1
-                while caller is None or caller.filename == caller_myself.filename:
-                    caller = getframeinfo(stack()[i][0])
-                    i += 1
+                while (
+                    any(caller.filename.endswith(x) for x in REMOVE_FROM_STACKTRACE)
+                    or MRBEAM_PLUGIN_SEARCH_STR not in caller.filename
+                ):
+                    try:
+                        caller = getframeinfo(stack()[i][0])
+                        self.logger.info(
+                            "ANDYTEST i: %s, filename: %s",
+                            i,
+                            caller.filename if caller else "noCallerObj",
```

Comment:
You have already insured `caller` is not `None` because your while loop uses `caller.filename` - ergo no need for 
`caller.filename if caller else "noCallerObj",` , just `caller.filename,`